### PR TITLE
Add coniks inclusion test vectors to merkle_test

### PR DESCRIPTION
**Pull request by gdbelvin**
The `merkle.MapVerifier` lacks any test vectors for the coniks hasher.  
To facilitate interoperability with JavaScript (coming soon!), I've added some test vectors for the CONIKS hash strategy.

### CODE REVIEW

File: merkle/map_verifier_test.go [hunk 1]
```diff
@@ -17,11 +17,51 @@ package merkle
...
+		t.Run(tc.desc, func(t *testing.T) {
+			if err := VerifyMapInclusionProof(tc.treeID, tc.index, tc.leaf, tc.root, tc.proof, h); err != nil {
+				t.Errorf("VerifyMapInclusionProof failed: %v", err)
+			}
+		})
+	}
+}
+
 func TestMapHasherTestVectors(t *testing.T) {
+	// Test vectors were copied from a python implementation.
```
Comment (phad) [inline merkle/map_verifier_test.go]: gawd help us.  Which one?

File: merkle/map_verifier_test.go [hunk 2]
```diff
@@ -17,11 +17,51 @@ package merkle
...
+			desc:   "Empty proof",
+			index:  []byte{0xa5, 0xd1, 0x37, 0xb4, 0x39, 0x38, 0xce, 0x3f, 0x28, 0x69, 0x55, 0x42, 0x6, 0xb1, 0x96, 0x4b, 0x84, 0x95, 0xda, 0xa2, 0x54, 0x8, 0xf2, 0x75, 0x75, 0x80, 0x1a, 0xc0, 0x71, 0xba, 0xed, 0xa7},
+			proof:  make([][]byte, 256),
+			root:   []byte{0x0e, 0xfc, 0x54, 0xad, 0xe0, 0xfc, 0xe8, 0x76, 0x55, 0x8c, 0x97, 0x38, 0xf5, 0xaa, 0x89, 0xe4, 0xd9, 0x9c, 0x0b, 0x8b, 0x6f, 0xe0, 0xb6, 0x2d, 0xbf, 0x63, 0x59, 0xcf, 0xc2, 0xad, 0xbb, 0xd7},
+			treeID: 9175411803742040796,
+		},
+		{
+			desc:  "One item",
+			index: []byte{0xb7, 0x57, 0x2d, 0xf6, 0xe1, 0x9, 0x1f, 0xc0, 0x6, 0x9e, 0x4, 0xbf, 0x80, 0x98, 0x75, 0x25, 0xe7, 0x7a, 0xc9, 0xa6, 0xc2, 0x94, 0xd2, 0x8d, 0xb7, 0xf4, 0xe3, 0x60, 0x25, 0x1d, 0x83, 0xbf},
+			proof: append(make([][]byte, 255), []byte{
```
Comment (phad) [inline merkle/map_verifier_test.go]: I always have to think when I read this code, but it's concise and I can't think of a nicer alternative.

File: merkle/map_verifier_test.go (hunk 1 above)
Comment (gdbelvin) [reply merkle/map_verifier_test.go]: I don't know! The first [commit](https://github.com/google/trillian/commit/c348bf22a236ec20c124ffaa6f4fa1122a3d81e5#diff-b9d1a46bce9eb9bc8d3203df5afc9345R99) has the simple comment `// Test data produced with python` 

@AlCutter do you remember anything about a map python implementation, perhaps lost to the sands of time?

File: merkle/map_verifier_test.go (hunk 2 above)
Comment (gdbelvin) [reply merkle/map_verifier_test.go]: ok

**Pull request merged**